src/curl.c

```diff
@@ -58,12 +59,57 @@ uint64_t total_curl_sz = 0;
 /* alternative CA Path */
 static char *capath = NULL;
 
+static bool spinner_callback_enable = false;
+
 /* Pretty print curl return status */
 static void swupd_curl_strerror(CURLcode curl_ret)
 {
 	error("Curl - Download error - (%d) %s\n", curl_ret, curl_easy_strerror(curl_ret));
 }
 
+static char spinner[5] = { '|', '/', '-', '\\'};
+static time_t begin = 0;
+
+static int progress_spinner_callback(void UNUSED_PARAM *clientp, double UNUSED_PARAM dltotal,
+			     double UNUSED_PARAM dlnow, double UNUSED_PARAM ultotal, UNUSED_PARAM double ulnow)
+{
+#define MAX_THRESHOLD_MULTIPLE_SIX 36000
+
+	if (begin ==0){
+		begin = time(NULL);
+	}
+	
+	if ((time(NULL) - begin) > 0.5) {
+		begin = time(NULL);
+		static int index;
+		info("[%c]", spinner[index % 6]);
+		fflush(stdout);
+		index++;
+		info("\b\b\b");
+		fflush(stdout);
+
+		/* reset index */
+		if (index == MAX_THRESHOLD_MULTIPLE_SIX){
+			index=0;
+		}
+
+	}
+
+	return 0;
+}
+
+void set_spinner_callback(bool onflag) {
```

Comment:
instead of a boolean here you can get the function that is going to be used for the callback. And instead of calling it set_spinner_callback() you can call it set_progress_callback(). The logic of the function can be the same.
The good thing about this change is that curl will be more generic to support any kind of progress callbacks that we may need in the future. What if we wan't to do more than just printing the spinner for other parts of the code?

Reply:
Then you can create a 

    void progress_set_spinner(bool onflag);

on progress.c that will call set_progress_callback(progress_spinner_callback) or set_progress_callback(NULL) to turn it on or off.

Reply:
Done